src/type/definition.js

```diff
@@ -506,6 +506,11 @@ function defineFieldMap<TSource, TContext>(
       `${type.name}.${fieldName} field type must be Output Type but ` +
       `got: ${String(field.type)}.`
     );
+    invariant(
+      isValidResolver(field.resolve),
```

Comment:
We should probably enable `== null` in the linter. I think eslint now has the ability to whitelist that particular kind of usage. It's equivalent to the longer form.